aptos-move/aptos-gas-meter/src/algebra.rs

```diff
@@ -18,7 +19,10 @@ use std::{fmt::Debug, ops::AddAssign};
 /// Base gas algebra implementation that tracks the gas usage using its internal counters.
 ///
 /// Abstract gas amounts are always evaluated to concrete values at the spot.
-pub struct StandardGasAlgebra {
+pub struct StandardGasAlgebra<'a, T>
+where
+    T: BlockSynchronizationKillSwitch + ?Sized,
```

Comment:
Hm, does `?Sized` actually make this dispatch dynamically? Why the compiler asks for it?

Reply:
From errors I'm getting I think it's actually still dynamic, but not sure it's because of ?Sized. Moreover, I eliminated it from ExecutorView requirements now that we added explicitly and to CodeStorage, but still doesn't allow me to remove ?Sized from here and prod_gas_meter requirements. I'll add a TODO and try to look underneath to figure out what's going on with generics under the hood.

Reply:
Fixed.